test/sqlite/sqllogic_test_logger.cpp

```diff
@@ -13,45 +18,85 @@ SQLLogicTestLogger::SQLLogicTestLogger(ExecuteContext &context, const Command &c
 SQLLogicTestLogger::~SQLLogicTestLogger() {
 }
 
-void SQLLogicTestLogger::Log(const string &str) {
-	std::cerr << str;
+void SQLLogicTestLogger::SafeAppend(const string &log_message) {
+	lock_guard<mutex> lock(summary_mutex);
+	failures_summary.push_back(log_message);
+}
+
+void SQLLogicTestLogger::LogBoth(const string &log_message) {
```

Comment:
Can we just rename this method to `LogFailure`? The fact that it writes to two destinations is more of a detail of the method.